src/lib.rs

```diff
@@ -13,3 +13,6 @@ mod traits;
 mod serde_impls;
 
 pub use crate::{range::TextRange, size::TextSize, traits::TextSized};
+
+#[cfg(target_pointer_width = "16")]
+compile_error!("text-size assumes usize >= u32 and does not work on 16-bit targets");
```

Comment:
When the goal is to compile on multiple cfgs, I definitely agree that specifying the behavior without `cfg` is better, if possible.

But for cases like this where some cfg isn't going to work at all, I don't agree that the array shenanigans are better.

To begin with, what we have on master is "[`cfn_assert!`](https://crates.io/crates/const_fn_assert)" and not "[`const_assert!`](https://crates.io/crates/static_assertions)", so we currently have panic at runtime (or error-by-default lint if it gets MIR const folded) behavior, rather than guaranteed compile error.

More importantly, though, is that `#[cfg] compile_error!` _exists_ so that cases like this can give a useful compile error message rather than just an opaque "const will always panic at runtime" error. And it's not like we have to worry about other cfgs with `usize < u32`; Rust nominally only supports down to a 16 bit `usize` (and a system with a pointer size of 256 values is constrained enough that good luck doing anything useful with anything other than asm).

We can do both (the `cfg` and `assert`), if you prefer, but the `cfg` pays for itself by providing a useful compile error and documentation of intent in this case.

Reply:
Alternatively, we can support 16 bit `usize` and only panic when we try to put a value `> usize::MAX` into `usize`. Personally, I think the clear message wins.

Reply:
>so we currently have panic at runtime (or error-by-default lint if it gets MIR const folded) 

This is a very good argument in favor of cfg, convinced!